resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java

```diff
@@ -117,7 +117,9 @@ public void close() {
         }
 
         for (final Map.Entry<EventExecutor, GenericFutureListener<Future<Object>>> entry : listeners) {
-            entry.getKey().terminationFuture().removeListener(entry.getValue());
+            if (entry.getKey().terminationFuture() != null) {
+                entry.getKey().terminationFuture().removeListener(entry.getValue());
```

Comment:
why this change ?

Reply:
We get null pointer exceptions otherwise. I'm not entirely sure why, it might have to do with our use of Thread.stop().

Reply:
hmm... this should never happen.